src/services/EcoBridge/Xdai/XdaiBridge.reducer.ts

```diff
@@ -0,0 +1,136 @@
+import { createSlice, EntityState, PayloadAction } from '@reduxjs/toolkit'
+import { TokenList } from '@uniswap/token-lists'
+
+import { BridgeTransactionStatus } from '../../../state/bridgeTransactions/types'
+import { BridgeDetails, BridgingDetailsErrorMessage, SyncState, XdaiBridgeList } from '../EcoBridge.types'
+import { xdaiBridgeTransactionAdapter } from './XdaiBridge.adapter'
+import { XdaiBridgeTransaction, XdaiMessage } from './XdaiBridge.types'
+
+interface XdaiBridgeState {
+  bridgingDetails: BridgeDetails
+  bridgingDetailsStatus: SyncState
+  lastMetadataCt: number
+  listsStatus: SyncState
+  lists: { [id: string]: TokenList }
+  transactions: EntityState<XdaiBridgeTransaction>
+}
+
+const initialState: XdaiBridgeState = {
+  transactions: xdaiBridgeTransactionAdapter.getInitialState({}),
+  bridgingDetails: {},
+  bridgingDetailsStatus: SyncState.IDLE,
+  lists: {},
+  listsStatus: SyncState.IDLE,
+  lastMetadataCt: 0,
```

Comment:
what is `lastMetadataCt`?

Reply:
`lastMetadataCt` is counter of `getBridgingMetadata` calls. It's way of cancelling request. 

### Example:
- user selects token and amount
- `getBridgingMetadata` is fetching details
- user changed amount or token
- `getBridgingMetadata` is executed once again
Note: (two calls of `getBridgingMetadata` are running)
- cancel first call
- display details for second call

```ts
// getBridgingMetadata method

// get lastMetadataCt from store
const requestId = this.store.getState().ecoBridge[this.bridgeId as XdaiBridgeList].lastMetadataCt

const helperRequestId = (requestId ?? 0) + 1

// if method is called, increment lastMetadataCt
this.store.dispatch(this.actions.requestStarted({ id: helperRequestId }))

// reducer

// if requestId is not equal lastMetadataCt and status is not `failed` then set bridgingDetailsStatus to `loading`
if (requestId !== state.lastMetadataCt) { 
  if (state.bridgingDetailsStatus === SyncState.FAILED) return
    state.bridgingDetailsStatus = SyncState.LOADING
    return
  } else {
    // if requestId and lastMetadataCt are the same then we want to display details
    state.bridgingDetailsStatus = SyncState.READY

    state.bridgingDetails.gas = gas
    state.bridgingDetails.fee = fee
    state.bridgingDetails.estimateTime = estimateTime
    state.bridgingDetails.receiveAmount = receiveAmount
}
```

Reply:
Thanks Mix1o for the super clean explanation! 🙌  Much clear now.